### PR TITLE
ABI healthcheck

### PR DESCRIPTION
**Pull request by theHamsta**
This should help users @przepompownia 

Ref: https://github.com/nvim-treesitter/nvim-treesitter/issues/1110#issuecomment-811775871

### CODE REVIEW

**Comment by vigoux** [comment]
Hmmm, not sure that's the way we should go... Maybe doing the check for ABI compatibility in Lua side for each parser would be better.

**Comment by theHamsta** [comment]
But the ABI check is already done (also shown in the individual error messages for each parser). The issue is that in case of a miss-match users are confused what to do. This part of the health-check is rather about our plugin providing the Git SHAs for 13 ABI. So the user should rather upgrade Neovim instead of downgrading the parsers.

In future, we can avoid ABI chaos by just versioning our lockfile. So when a new ABI comes our we will just open a new section in our lockfile with the new ABI version while the old one remains unchanged and can still be used without problems.

File: lua/nvim-treesitter/health.lua [hunk 1]
```diff
@@ -25,6 +27,17 @@ local function install_health()
...
+  if fn.executable('node') == 0 then
+    health_warn('`node` executable not found (only needed for :TSInstallFromGrammar,'..
+                ' not required for :TSInstall)')
+  else
+    local handle = io.popen('node --version')
+    local result = handle:read("*a")
+    handle:close()
+    local version = vim.split(result,'\n')[1]
+    health_ok('`node` found '..version..' (only needed for :TSInstallFromGrammar)')
+  end
```
Comment (kyazdani42) [inline lua/nvim-treesitter/health.lua:39]: thinks i've seen that in another PR :)

**Review by kyazdani42 — COMMENTED** [review]
hum looks like this PR has some node check that is in another PR you've made, could you either close the other or fix this one :) ?

File: lua/nvim-treesitter/health.lua (hunk 1 above)
Comment (theHamsta) [reply lua/nvim-treesitter/health.lua:39]: rebased! :smile:

Comment (theHamsta) [reply lua/nvim-treesitter/health.lua:39]: not yet. remove the wrong commits in the wrong PR

**Review by kyazdani42 — APPROVED** [review]
:+1:

**Review by vigoux — APPROVED** [review]

**Pull request merged**